WIP | Select Domain Registration Period From Dropdown

sharpirate opened the pull request:
Dropping this PR in favor of https://github.com/Automattic/wp-calypso/pull/80216 due to rebasing issues

sharpirate committed:
select domain registartion period from dropdown

sharpirate committed:
select domain registartion period from dropdown

sharpirate committed:
mock server cart

sharpirate committed:
mock server cart

dkd903 commented:
@Sharpirate  - I created a patch for the BE to support the multi year volumes. Can you give it a go please? https://code.a8c.com/D116538

![Screen Shot 2023-07-20 at 7 53 24 AM](https://github.com/Automattic/wp-calypso/assets/233879/d57e2997-7c8d-4261-ba10-adca3d2f7c24)

sharpirate committed:
display all term plans

sharpirate committed:
display all term plans

sharpirate committed:
determine variant selection based on volume for multi-year domains

sharpirate committed:
determine variant selection based on volume for multi-year domains

kangzj committed:
Odyssey Stats: fix page base when changed (#80066)

* fix page base when changed for Odyssey

* comment

kangzj committed:
change page page after replacing state (#80067)

mreishus committed:
FETCH_HEAD: Remove (#80063)

p-jackson committed:
Fix launchpad WSOD when the site has an unrecognised plan (#80013)

This situation would occur when the site has a plan that's not in the
@automattic/calypso-products package for whatever.

ddc22 committed:
Provide default value when domain form is not available (#80070)

worldomonation committed:
CI: add retry to Desktop App E2E as well as filter based on changed file path criteria. (#79903)

* .teamcity/_self/projects/DesktopApp.kt
- add retry to Desktop App E2E.
- add exclude trigger to filter out unnecessary builds.

* _self/projects/DesktopApp.kt
- import

* .teamcity/_self/projects/DesktopApp.kt
- remove retry within build step

Commit:
Domain transfer: Improve mobile text alignment (#80040)

yashwin committed:
Jetpack Pro Dashboard: implement rules to show one JITM at a time on the dashboard (#79510)

* Implement rules to show one JITM at a time on the dashboard

* Address PR comments

* Make banner logic to util function instead of a hook

* Rename file name to show-banner.ts

* Update show-banner util to return JSX.Element

mmtr committed:
Limited GS: Update stickers notes (#80048)

Updates the blog stickers added as part of the Limited Globals Styles logic so they are less confusing for HEs:
- The current user is no longer tracked. This was causing HEs to think they were accidentally adding these stickers.
- Added a "Automated sticker" to the sticker notes to make clear the sticker has not been added manually.

arthur791004 committed:
Assembler: Clean up the feature flag - pattern-assembler/logged-in-showcase (#80081)

yashwin committed:
Jetpack Pro Dashboard: show link with monitor paid upgrade cost (#80074)

* Move get products API call to SitesOverview & add it to the context API

* Create UpgradeLink component

* Show upgrade link for SMS notifications

* Show upgrade link for paid durations

* Show upgrade link for add email address.

* Hide upgrade link for bulk edit & fix naming conventions

arthur791004 committed:
Assembler: Clean up the feature flag - pattern-assembler/all-patterns-category (#80084)

robertsreberski committed:
Add Jetpack AI Yearly supporting code (#79959)

* Remove MONTHLY_ONLY code logic

* Add jetpack_ai_yearly to product constants

* Add jetpack_ai_yearly to product list, copy, icons

jjolmo committed:
Hide the go to campaign button if android version >= 22.9 (#79966)

* Hide the go to campaign button if android version >= 22.9

* Change the way we obtain device info

arthur791004 committed:
Assembler: Add the shuffle button to the action bar (#80078)

cpapazoglou committed:
Marketplace: updates copy and aligns CTA (#80031)

* Marketplace: updates copy

* Marketplace Thank You: aligns CTA link

---------

Co-authored-by: cpap <papazoglou.charalampos@gmail.com>

sdnunca committed:
I18N: Localize the Stats module names (#80079)

Co-authored-by: sdnunca <sorin@php.net>

klimeryk committed:
Show support links in Help Center instead of modal (#79741)

As suggested in pbvpgB-3o7-p2

If a user clicks on a support doc link in Calypso, a new modal (or tab) is opened with it. This takes the user away from the context they’re need that information the most. Thanks to Vertex's work on Help Center, we can instead show those pages the Help Center. This will keep the user in the Calypso context, while giving them a better UX. The floating window of Help Center can be moved around and stays as they navigate Calypso and, for example, follow the steps on the support doc.

I've used the existing InlineSupportLink component that is used across Calypso (though, not consistently). It has been introduced in 2018 in #26347, and fits perfectly our needs. In future patches, we can update any existing "normal" support links to use this component as well.

ouikhuan committed:
Migration Trial: Add migration notice to the manage purchase panel (#80029)

jornp committed:
i18n: Fix loading translations of devices in preview toolbar (#80085)

atanas-dev committed:
Jetpack Pro Dashboard: Move Backup Add-ons to their own section on the license screen (#80047)

paulopmt1 committed:
Domain Transfer: Update button.is-link style for stepper flows (#80061)

* Updated button.is-link style for stepper flows

* Updated hover and active colors

sdnunca committed:
Plans: Escape percent in plan features (#80093)

Co-authored-by: sdnunca <sorin@php.net>

mtias committed:
Domain Transfer: open step 3/4 by default (#80043)

danielbachhuber committed:
Hosting: Allow user to switch to 'Beta' version in staging (#80044)

---------

Co-authored-by: Philip Jackson <p-jackson@live.com>
Co-authored-by: Philip Jackson <p-jackson@users.noreply.github.com>

markbiek committed:
Switch the plugin-install check to use exponential backoff (#79797)

We've been getting some timeout errors at this step. The timeout is
already quite high (5m) so we've added some additional error logging and
switched how we wait between checks to use an exponential backoff
approach.

allilevine committed:
Remove shouldShowConfettiExplosion. (#80064)

sirbrillig committed:
Checkout: Move 3DS challenge code to own functions (#79946)

* Move 3DS challenge code to own functions

* Log payment intent ID to logstash

* Fix typeof check

* Use doesTransactionResponseRequire3DS for other responses

* Log failed 3DS errors

* Change logstash level to info

* Allow passing tags to logstash

* Add tags to each new logStashEvent

* Record tags as a single string

* Pass tags as strings

* Mock logstash endpoint in card tests

ivan-ottinger committed:
Subscribers Page: Code cleanup (feature flag and translation fallback) (#80041)

* Remove `subscribers-page-new` feature flag and related code

* Remove translation fallbacks

* Remove `people-add-subscribers`

jboland88 committed:
Update styles for pricing page nav (#79974)

* Update styles for pricing page nav

* Fix some spacing inconsistency

CodeyGuyDylan committed:
Add akismet enterprise benefits to checkout (#80052)

* Add Feature strings for Akismet Enterprise plans

* Add benefits for Akismet Enterprise in checkout

* Update unit tests

valterlorran committed:
Add share site modal to the newsletter task lists (#80065)

lsl committed:
Add presales chat to checkout and thank you page (#80075)

Co-authored-by: Candy Tsai <candy02058912@gmail.com>
Co-authored-by: Omar Alshaker <omar@omaralshaker.com>

DustyReagan committed:
Change copy tense to 'We’ll pay for an extra year' before checkout (#80098)

valterlorran committed:
Launchpad: Add task counter (#79798)

* Remove files not related to this change

* Fix spacing

* Show the counter when the task is completed

* Align the counter text with the task name, when the task is complete

ebinnion committed:
Domains: When a domain is registered prompt user to transfer (#79802)

* Domains: When a domain is registered prompt user to transfer

* Domains: Update notice to prefill transfer domain when coming from domain only flow

WBerredo committed:
Add and update the interval param when changing the billing switcher (#79969)

dlind1 committed:
Wrap a string in translate function (#80006)

noahtallen committed:
Update to typescript 5.1 (#74540)

* Update typescript to 5.1

* Update @typescript-eslint to latest v5 release to support ts 5.1

* Update popperjs to fix type issue

* Update emotion for compatibility with Typescript 5

* Fix the GlobalStylesContext types for typescript 5 (#79722)

* Fix unspecific type parameter to isObjKey

* Fix implementation details for Report

* Fix BaseButton type issue

- Use wrapper fn to handle issues with isValidElement

* Use type parameter to fix cloneElement issues

* Fix issue where Typescript doesn't infer generic correctly
- This applies to createSelector and treeSelect

* Add wrapper type to handle error catch

* Fix another HoC OuterProps issue

danielbachhuber committed:
Hosting: Add a relative link to the Staging Site module (#80117)

Commit:
Add core-profiler specific events on Jetpack connection page (#79892)

* Remove unused import

* Add connect your account button track for wc core profiler

* Record user account creation event for core profiler

* Rename calypso_jpc_different_user_click to calypso_jpc_wc_coreprofiler_different_user_click

roo2 committed:
Fix issue hiding blogging prompts card when on the site checklist view on my home (#79776)

* dismiss single and multiple card views differently

* fix comment

* fix name of post_launch view

yashwin committed:
Jetpack Pro Dashboard: implement rules to show one JITM at a time on the Licenses page (#80086)

kangzj committed:
refactor conditions for better readibility (#80072)

kangzj committed:
change learn more links (#80123)

Commit:
Remove SVN connection, and switch Jetpack staging e2e to finish build trigger. (#80113)

okmttdhr committed:
Remove unused methods in Theme Detail page (#80129)

okmttdhr committed:
Create selectors for the Live Preview feature (#80017)

miksansegundo committed:
Use creatio theme demo site to preview posts patterns (#80128)

okmttdhr committed:
Add the “Live Preview” option on the Themes Showcase page (#80071)

jjolmo committed:
Call upsert method if the DSP call to get campaigns fail (#80100)

jjolmo committed:
Remove the breadcrumb if jetpack in campaign list (blazepress) (#79832)

* Remove the breadcrumb if jetpack

* Change ther way we hide the breadcrumb and fix margin error in jetpack

* Change ther way we hide the breadcrumb and fix margin error in jetpack

* remove padding when app mode in blazepress campaign details

sbarbosa committed:
Fixes the contact support link in the Blaze Dashboard app (#80111)

WBerredo committed:
Signup: Add the plans-business-with-plugin-step (#80105)

* Signup: Add the plans-business-with-plugin-step

* Update step name

paulopmt1 committed:
Domain transfer: Update google domain transfer helper copy (#80106)

* Updated step 2 copy on google domain transfer

* Added conditional use of the new text since we already have some translations of it

* Copy updated

j6ll committed:
Remove this flag, no longer required (#79878)

Co-authored-by: James Gill <james.gill@automattic.com>

worldomonation committed:
client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domain-code-pair.tsx (#80132)

- update the link to the right anchor on the support page.

daledupreez committed:
Launchpad: Add client-side handling for calypso_path task data (#80090)

* Launchpad: Add client-side handling for calypso_path task data

* Remove case for manage_subscribers

* Make uiContext optional

* Reinstate custom handling and add logging for that code path

ddc22 committed:
GF Signup: Add a modal upsell when a user does not chose a custom domain and choses a free plan (#79937)

* Add a modal and feature flag when selected free domain

* Fix cosmetic issues

* Provide custom domain suggestion

* Fix price issues

* Do domain search with last searched string

* Remove dependence on progress store

* Disable feature in wpcalypso

* Fix type issue and lazy loading text

* Fix color type

* Fix a few cosmetic changes and wordings

* Cleanup last domain searched

* Fix yourgroovydomain bold and italic

* Remove brackets

* Improve deprecated message

* Use proper dotcom domain pricing from API, Use dynamic plan title instead of hardcoding, Use correct plan prices which includes discounts, Add comment to clarify free free modal

* Rename flag

* Remove unnecessary hook

* Generate subdomain suggestion when suggestion not provided

* Fix 0 prizes

* Fix descripency in domain when user already signed up, remove knowledge of freeSubdomain from the modal

* Change component name

sdnunca committed:
I18N: Update My Stats copy plural (#80148)

Co-authored-by: sdnunca <sorin@php.net>

Commit:
Plugins: Update how translations are handled in the action modal (#79135)

jeyip committed:
Plans 2023: Implement storage add on dropdown (#79041)

* Add storage upgrade dropdown behind feature flag

* Pass feature flag as props through plans features main

bogiii committed:
Fix call undefined function (#80091)

bogiii committed:
Migration trial: Import flow: Create migration trial step (#80023)

* Import flow: Create migration trial step

* Trial migration: add unsupported features list block

* Trial migration: Create business features list modal

* Button container: add border top line

* Migration trial: Add navigation back handler

kangzj committed:
Stats: Show new title for sites with free stats (#80125)

* show different copy for free stats

* Update all notices and simplify the logic

* Restore inlined notice args (for smaller diff)

---------

Co-authored-by: Jason Moon <jsnmoon@users.noreply.github.com>

jeyip committed:
Storage Add-Ons: Display the storage label for multi-year plans (#80155)

* Displays the storage label for multi-year plans

kangzj committed:
hide upgrade notice on success notice shown (#80124)

jjchrisdiehl committed:
Billing: Remove taxamo external link (#80119)

* Remove taxamo external link

* Remove ReceiptExternalLink component

---------

Co-authored-by: Chris Diehl <jjchrisdiehl@gmail.com>

zaguiini committed:
Hosting onboarding: add padding to the bottom of the FAQ in the plans step (#80154)

zaguiini committed:
Hosting onboarding: even spacing between elements of the plans step (#80153)

kangzj committed:
Paid Stats: default commercial plan for sites with personal plan already (#80161)

* default commercial plan for sites with personal plan already

* fix wrong import

* revert unnecessary changes

* remove obsolete comment

* export everything together

* add default values

* add comment

lupus2k committed:
Domains manage: Link site name to site's /home (#80159)

kangzj committed:
Paid Stats: Starting pricing from non-zero for slider when user already has stats free (#80162)

* starting from non zero when user already has stats free

* Update client/my-sites/stats/stats-purchase/index.tsx

Co-authored-by: Dognose <dognose24@users.noreply.github.com>

---------

Co-authored-by: Dognose <dognose24@users.noreply.github.com>

okmttdhr committed:
Avoid showing the Live Preview for unsupported themes (#80163)

wojtekn committed:
Jetpack Error UX: Add component that checks Jetpack connection health (#79965)

* Add component that checks Jetpack connection health

* Call new reducer when jetpack jitm request fails

* Add another reducer for handling helathy connection

* Add feature variable to other environments

* Add own state for jetpack connection health status

* Fix formatting

* Implement reducer action on healthy connection

* Fix types handling

* Add tests and readme for the new state

* Add tests for a Jetpack connection health selector

* Remove unnecessary function

* Link notice action to page that can help user with Jetpack connection

* Change warning to error to follow designs

* Name state variables consistantly

* Save unhealthy connection error with a state

* Move onsuccess handler to the hook

* Add test for new state reducer action

* Add tracks for Jetpack connection notice view and click

* Enable new UI on test environment

* Use better way to log component view using track

* Update Jetpack connection health error text

* Enable Jetpack connection health notice for all environments

---------

Co-authored-by: Daniel Bachhuber <daniel.bachhuber@automattic.com>

yoavf committed:
Don't prefetch the GPT response on the form screen, only fetch it when the user clicked 'continue' (#80142)

dkd903 committed:
Merge branch 'add/multi-year-domains' of github.com:Automattic/wp-calypso into add/multi-year-domains

zaguiini committed:
Hosting onboarding: use custom heading and subheading copies in the plans step (#80152)

miksansegundo committed:
Virtual Themes - Don't copy starter content from Creatio demo site (#80135)

* Not use Creatio demo site posts to copy site starter content

* Ensure the starter content is trimmed in case sourceSiteId is defined

sharpirate committed:
use isMultiYearDomainProduct to do comparison by volume

sharpirate committed:
remove mockResponse
